public/app/percona/shared/helpers/api.ts

```diff
@@ -61,3 +62,47 @@ export const apiManagement = new ApiRequest({ baseURL: '/v1/management' });
 export const apiInventory = new ApiRequest({ baseURL: '/v1/inventory' });
 export const apiSettings = new ApiRequest({ baseURL: '/v1/Settings' });
 export const isApiCancelError = (e: any) => axios.isCancel(e);
+
+export const translateApiError = (error: ApiErrorCode): ApiVerboseError | undefined => {
+  const map: Record<ApiErrorCode, ApiVerboseError> = {
+    [ApiErrorCode.ERROR_CODE_INVALID_XTRABACKUP]: {
+      message: 'Different versions of xtrabackup and xbcloud.',
+      link: '',
+    },
+    [ApiErrorCode.ERROR_CODE_XTRABACKUP_NOT_INSTALLED]: {
+      message: 'Xtrabackup is not installed.',
+      link: 'https://www.percona.com/doc/percona-xtrabackup/8.0/installation.html',
+    },
+    [ApiErrorCode.ERROR_CODE_INCOMPATIBLE_XTRABACKUP]: {
+      message: 'Xtrabackup version is not compatible.',
+      link: 'https://www.percona.com/doc/percona-xtrabackup/8.0/installation.html',
+    },
+    [ApiErrorCode.ERROR_CODE_INCOMPATIBLE_TARGET_MYSQL]: {
+      message: 'Target MySQL version is not compatible.',
+      link: 'https://www.percona.com/doc/percona-xtrabackup/8.0/installation.html',
+    },
+  };
+
+  const translatedError = map[error];
+
+  return translatedError;
+};
+
+export const apiErrorParser = (e: AxiosError): ApiVerboseError[] => {
+  const errorData: ApiError = e.response?.data;
+  const result: ApiVerboseError[] = [];
+
+  if (errorData) {
+    const { details = [] } = errorData;
+
+    for (let i = 0; i < details.length; i++) {
```

Comment:
Can it be refactored to a `reduce`?  So it uses ES6 syntax and avoids push.

Reply:
Here we are returning an array, isn't `reduce` normally used to take an array and return a single value instead?
I guess we can't also use `map`, as some values have to be skipped if undefined (unless we did `.map().filter()`)